onnx/reference/ops/op_cum_prod.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) ONNX Project Contributors
+
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+from onnx.reference.op_run import OpRun
+
+
+class CumProd(OpRun):
+    def _run(self, x, axis, exclusive=None, reverse=None):  # type: ignore
+        if not isinstance(axis, (np.int32, np.int64)):  # type: ignore
+            if len(axis.shape) > 1 or (len(axis.shape) > 0 and axis.shape[0] != 1):  # type: ignore
+                raise RuntimeError(
+                    f"axis must be an array of one number not {axis} (shape {axis.shape})."  # type: ignore
+                )
+            if len(axis.shape) > 0:  # type: ignore
+                axis = axis[0]
```

Comment:
How about this:

```suggestion
        # coerce NumPy generics into array object of rank-0
        axis = np.asarray(axis)
        if axis.ndim != 0:
            raise RuntimeError("'axis' input must be a rank-0 tensor, got `{axis.ndim}`")
```

`RuntimeError` is actually the wrong exception type, but appears to be used like this throughout this project.

Reply:
Happy to do a pass to update the code base. What would be the correct exception?

Reply:
This should be ValueError right?

Reply:
If in doubt it usually is ;)